### PR TITLE
Setup a mechanism to update cached filter data

### PR DESCRIPTION
**Pull request by jaladh-singhal**
This PR aims to add a mechanism that will check the time it has been since last download, to inform user by automatically logging a message to update the filter data. So that user can use `update_filter_data()` to update the cache on their disk.

The tasks under this PR can be summarized as:
- [x] Devise logic to update the cached data & create a function for it in working condition
- [x] Setup a reminding log to update the cache if >1 month from last download
- [x] Identify further fail cases (if any) & add code for exception handling 
- [x] Create tests for the added functions
- [x] Update documentation to show how to update the cache

### CODE REVIEW

File: wsynphot/config.py [hunk 1]
```diff
@@ -1,43 +1,119 @@
...
 import yaml
+from datetime import datetime
 
 from astropy.config import get_config_dir
 
+
 WSYNPHOT_PATH = WSYNPHOT_PATH[0]
 DEFAULT_CONFIG_PATH = os.path.join(WSYNPHOT_PATH, 'data', 'default_wsynphot_config.yml')
 DEFAULT_DATA_DIR = os.path.join(os.path.expanduser('~'), 'Downloads', 'wsynphot')
+CONFIG_FPATH  = os.path.join(get_config_dir(), 'wsynphot_config.yml')
```
Comment (wkerzendorf) [inline wsynphot/config.py]: there should be a function that get the config path

Comment (wkerzendorf) [reply wsynphot/config.py]: actually - that looks fine - disregard.

File: wsynphot/config.py [hunk 2]
```diff
@@ -1,43 +1,119 @@
...
+        yaml.dump(config, open(CONFIG_FPATH, 'w'), default_flow_style=False)
         data_dir = DEFAULT_DATA_DIR
 
     if not os.path.exists(data_dir):
         raise IOError('Data directory specified in {0} does not exist'.format(data_dir))
 
     return data_dir
+
+
+def get_cache_dir():
```
Comment (wkerzendorf) [inline wsynphot/config.py]: docstrings !! 😉

File: wsynphot/config.py [hunk 3]
```diff
@@ -1,43 +1,119 @@
...
+
+
+def get_calibration_dir():
+    calibration_dir = os.path.join(get_data_dir(), 'calibration')
+    if not os.path.exists(calibration_dir):
+        os.makedirs(calibration_dir)
+    return calibration_dir
+
+
+def get_cache_updation_date():
```
Comment (wkerzendorf) [inline wsynphot/config.py]: returns is missing in docstring

File: wsynphot/config.py [hunk 4]
```diff
@@ -1,43 +1,119 @@
...
+    to be used by cache download/update functions for saving when they were 
+    called last time"""
+
+    config = get_configuration()
+    current_date = datetime.now().date()
+    config['cache_updation_date'] = str(current_date)
+    yaml.dump(config, open(CONFIG_FPATH, 'w'), default_flow_style=False)
+
+
+def rectify_cache_updation_date(config, error_log):
```
Comment (wkerzendorf) [inline wsynphot/config.py]: returns is missing

File: wsynphot/config.py (hunk 2 above)
Comment (jaladh-singhal) [reply wsynphot/config.py]: Okay, I thought it's config file so it's okay to leave/minimize docstrings :smile:

**Pull request merged**